src/json_node.rs

```diff
@@ -0,0 +1,240 @@
+use crate::select::select_value::{SelectValue, SelectValueType, ValueUpdater};
+use crate::select::JsonPathError;
+use crate::select::SelectorMut;
+use serde_json::map::Entry;
+use serde_json::Value;
+
+impl SelectValue for Value {
+    fn get_type(&self) -> SelectValueType {
+        match self {
+            Value::Bool(_) => SelectValueType::Bool,
+            Value::String(_) => SelectValueType::String,
+            Value::Null => SelectValueType::Null,
+            Value::Array(_) => SelectValueType::Array,
+            Value::Object(_) => SelectValueType::Dict,
+            Value::Number(n) => {
+                if n.is_i64() {
+                    SelectValueType::Long
+                } else if n.is_u64() {
+                    SelectValueType::Long
+                } else if n.is_f64() {
+                    SelectValueType::Double
+                } else {
+                    SelectValueType::Undef
+                }
```

Comment:
What is the purpose of `Undef`? Can it ever happen that the value of the number is neither of `i64/u64/f64`? It seems like it cannot, so it would be cleaner to simply `panic!()` here and get rid of the `Undef` value altogether.

Reply:
Removed